### PR TITLE
Move over TokenizedDerivative from v0/ to core/

### PR DESCRIPTION
**Pull request by ptare**
Issue #453 

Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

File: core/contracts/ContractCreator.sol [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import "./Finder.sol";
+import "./Registry.sol";
+
+
+// TODO(ptare): Make this (and all contracts) Withdrawable.
+contract ContractCreator {
+    address internal finderAddress;
+    address internal adminAddress;
+    address internal priceFeedAddress;
```
Comment (rawfalafel) [inline core/contracts/ContractCreator.sol]: This looks like copy/pasta from `v0`, but I'm having a hard time understanding why `adminAddress` and `priceFeedAddress` might belong in this contract, and what this contract's responsibilities are at a high level.

File: core/contracts/TokenizedDerivative.sol [hunk 2]
```diff
@@ -7,15 +7,16 @@ pragma solidity ^0.5.0;
 
 pragma experimental ABIEncoderV2;
 
-import "./AdminInterface.sol";
+import "./AdministrateeInterface.sol";
```
Comment (rawfalafel) [inline core/contracts/TokenizedDerivative.sol]: :laughing: I'm slightly offended by this name.

**Comment by ptare** [comment]
> Seems like there's a shift from storing `*Interface` objects as instance variables to abstracting that into a `Finder` class which maps strings to addresses. But `PriceFeedInterface` and `ReturnCalculatorInterface` aren't stored in the `Finder` class. Why's that?
> 
> I'm nitpicking/trying to understand how we want to structure this contract. This is awesome though.

I think the idea behind the Finder class was to allow for upgrades of Oracle components (i.e., "proxying"). While the Registry, Store, and Voting are part of the Oracle, I considered the PriceFeed and ReturnCalculator to be separate.

Thoughts?

**Comment by ptare** [comment]
> > > Seems like there's a shift from storing *Interface objects as instance variables to abstracting that into a Finder class which maps strings to addresses. But PriceFeedInterface and ReturnCalculatorInterface aren't stored in the Finder class. Why's that?
> > > I'm nitpicking/trying to understand how we want to structure this contract. This is awesome though.
> > 
> > 
> > I think the idea behind the Finder class was to allow for upgrades of Oracle components (i.e., "proxying"). While the Registry, Store, and Voting are part of the Oracle, I considered the PriceFeed and ReturnCalculator to be separate.
> > Thoughts?
> 
> TLDR: I think we should add the `PriceFeed` to the `Finder`, but not add the `ReturnCalculator`.
> 
> Different leverages => different return calculators with the current design, so since the system will inevitably have multiple return calculators, storing them in the `Finder` would force us to make the `Finder` much more complex. I also don't see any reason that the `ReturnCalculator` would need to be upgraded since it requires no voter or UMA input or maintenance, so contracts can depend on it for as long as they see fit.
> 
> As for the price feed, it's a little more ambiguous because much of the system is currently built around a singleton price feed, but we were initially designing around the idea of people being able to spin up their own price feeds to back their contracts if they so chose. However, the price feed does require maintenance on our end, so it might be painful to continue to operate multiple price feeds because we're unable to force an upgrade. I could see the `Finder` being useful to point to the official UMA `PriceFeed` and, if necessary, adding an option to allow people to bring their own and hardcode an address.

Done (happy to discuss more offline). Do you think it also makes sense to push `adminAddress` to the Finder and eliminate `ContractCreator` in this PR?

File: core/contracts/ContractCreator.sol [hunk 3]
```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.5.0;
```
[the remark this answers is not in the document]
Comment (ptare) [reply core/contracts/ContractCreator.sol]: I'm not sure, might be that there were too many changes as you suggested.

[the remark this answers is not in the document]
Comment (mrice32) [reply core/contracts/ContractCreator.sol]: Yup, probably so. There's nothing you can really do to correct it since git either automatically picks it up or it doesn't.
